libp2p/ENR.cpp

```diff
@@ -0,0 +1,148 @@
+// Aleth: Ethereum C++ client, tools and libraries.
+// Copyright 2019 Aleth Authors.
+// Licensed under the GNU General Public License, Version 3.
+
+#include "ENR.h"
+#include <libdevcore/SHA3.h>
+
+static std::string const c_keyID = "id";
```

Comment:
I don't think it is great to have global dynamically constructed objects.

https://godbolt.org/z/qZwVxl

Reply:
Because of run-time initialization cost? Is it better to have dynamic allocation each time this code is run? (`string` is probably short-string-optimized, but `vector` isn't)

Reply:
Yes, to avoid dynamic initialization during startup.

This can land, but I'm leaving it to consider changes.